server/routes/referrals/desiredOutcomesForm.ts

```diff
@@ -0,0 +1,29 @@
+import { Request } from 'express'
+import { DraftReferral } from '../../services/interventionsService'
+import { DesiredOutcomesError } from './desiredOutcomesPresenter'
+
+export default class DesiredOutcomesForm {
+  private constructor(private readonly request: Request) {}
+
+  static async createForm(request: Request): Promise<DesiredOutcomesForm> {
+    return new DesiredOutcomesForm(request)
+  }
+
+  get paramsForUpdate(): Partial<DraftReferral> {
+    return {
+      desiredOutcomesIds: this.request.body['desired-outcomes-ids'],
+    }
+  }
+
+  get isValid(): boolean {
+    return this.request.body['desired-outcomes-ids'] !== null && this.request.body['desired-outcomes-ids'] !== undefined
+  }
+
+  get error(): DesiredOutcomesError | null {
+    if (this.isValid) {
+      return null
+    }
+
+    return { message: 'Select desired outcomes' }
```

Comment:
We've been using the `errorMessages.ts` file for all of our validation messages. I'll move this there.